### DISCUSSION TITLE
GOKIT Real Work Examples :)

### DISCUSSION THREAD

**Discussion by kolbis** [go-kit/kit#1200] [Show and tell]
Hi,
Since I like the work which was done in go-kit, i wanted to contribute back.

I created two repositories:
1. [corego](https://github.com/kolbis/corego)
 
GO KIT is one of the popular toolkit for building microservices in GO. It does a lot of heavy lifting when you get started. Yet, while working with Go Kit, I found there are still missing important concepts, such as configuration management, transporting context between microservices, working with databases, handling execution fallbacks and more. corego tries to be more opinionated and focus on consul and coreDNS for service discovery, consul as, remote KV, zap as logger, zipkin for tracing, cassandra, elastic, mongo for databases and rabbit for amqp transport

2. [go kit user microservice](https://github.com/kolbis/go-kit-user-ms-example)
GO-KIT is one of the popular toolkit for building microservices in GO. Yet, it is still pretty cumbersome and hard to get started with. GO-KIT-USER-MS-EXAMPLE repo aims to help those who are considering using GO-KIT.
Rabbit Transports, Rest Transport, Consul and coreDNS and more.

I plan to keep on maintaining these repos and update them as I proceed my journey with go-kit

**Comment by sagikazarmark** [comment]
Nice work!
You might be interested in https://github.com/sagikazarmark/modern-go-application covering similar concepts.
